be/src/http/action/update_config_action.cpp

```diff
@@ -316,6 +317,16 @@ Status UpdateConfigAction::update_config(const std::string& name, const std::str
             }
             return Status::OK();
         });
+        _config_callback.emplace("load_spill_merge_memory_limit_percent", [&]() -> Status {
+            // The change of load spill merge memory will be reflected in the max thread cnt of load spill merge pool.
+            int64_t load_spill_merge_memory_limit_bytes = GlobalEnv::GetInstance()->process_mem_tracker()->limit() *
+                                                          config::load_spill_merge_memory_limit_percent / (int64_t)100;
+            int max_merge_blocks_thread = load_spill_merge_memory_limit_bytes / config::load_spill_max_merge_bytes;
+            int cpu_cores = CpuInfo::num_cores();
+            int max_threads = std::max<int>(1, std::min<int>(max_merge_blocks_thread, cpu_cores));
+            return StorageEngine::instance()->load_spill_block_merge_executor()->get_thread_pool()->update_max_threads(
+                    max_threads);
+        });
```

Comment:
redundant code, better refactor to a common function like calc_load_spill_merge_thread_count(int load_spill_merge_memory_limit_percent)

Reply:
done